language/compiler/bytecode_source_map/src/mapping.rs

```diff
@@ -0,0 +1,486 @@
+// Copyright (c) The Libra Core Contributors
+// SPDX-License-Identifier: Apache-2.0
+
+use crate::source_map::{ModuleSourceMap, SourceName};
+use failure::prelude::*;
+use libra_types::identifier::IdentStr;
+use std::collections::{BTreeMap, HashMap};
+use vm::access::ModuleAccess;
+use vm::file_format::{
+    CodeOffset, CompiledModule, CompiledScript, FieldDefinitionIndex, FunctionDefinition,
+    FunctionDefinitionIndex, SignatureToken, StructDefinition, StructDefinitionIndex,
+    StructFieldInformation, TableIndex, TypeSignature,
+};
+
+/// An object that associates source code with compiled bytecode and source map.
+#[derive(Debug)]
+pub struct SourceMapping {
+    // The various options that we can set for disassembly.
+    options: SourceMappingOptions,
```

Comment:
Do we need these both here and in the disassembler `main`? It seems like having the options live only in the `main` and having the caller pass them into the disassemble logic would be more functional/easier to understand.

Reply:
I wanted to decouple the `main` from the implementation of the disassembler -- since we also want to support calling this from within Rust itself.